pkg/filesystem/subvolume.go

```diff
@@ -155,7 +240,16 @@ func getSubvolumeState(ctx context.Context, clientsets *k8sutil.Clientsets, oper
 
 // gets list of filesystem
 func getFileSystem(ctx context.Context, clientsets *k8sutil.Clientsets, operatorNamespace, clusterNamespace string) ([]fsStruct, error) {
-	fsList, err := exec.RunCommandInOperatorPod(ctx, clientsets, "ceph", []string{"fs", "ls", "--format", "json"}, operatorNamespace, clusterNamespace, true)
+
+	cmd := "ceph"
+	args := []string{"fs", "ls", "--format", "json"}
+
+	if checkForExternalStorage(ctx, clientsets, clusterNamespace) {
+		m, admin_id, admin_key := getExternalClusterDetails(ctx, clientsets, operatorNamespace, clusterNamespace)
+		args = []string{"fs", "ls", "--format", "json", "-m", m, "--id", admin_id, "--key", admin_key}
+	}
```

Comment:
why cant we run this from the operator pod? i hope rook also runs ceph commands form the operator pod. @subhamkrai can you please check this.

Reply:
Not sure what exactly is your query, we are running from operator pod only, this is just to get the arguements.